lib/src/view/broadcast/broadcast_game_screen.dart

```diff
@@ -112,19 +112,19 @@ class _BroadcastGameScreenState extends ConsumerState<BroadcastGameScreen>
           ),
         ],
       ),
-      body: switch ((broadcastRoundGameState, broadcastGameState)) {
-        (AsyncData(), AsyncData()) => _Body(
-          widget.tournamentId,
-          widget.roundId,
-          widget.gameId,
-          widget.tournamentSlug,
-          widget.roundSlug,
-          tabController: _tabController,
-        ),
-        (AsyncError(:final error), _) => Center(child: Text('Cannot load broadcast game: $error')),
-        (_, AsyncError(:final error)) => Center(child: Text('Cannot load broadcast game: $error')),
-        _ => const Center(child: CircularProgressIndicator.adaptive()),
-      },
+      body:
+          (broadcastRoundGameState.hasValue && broadcastGameState.hasValue)
```

Comment:
I think we should further improve and simplify this:

- instead of watching 2 providers, combine them into a single one, in order to have a `AsyncData<(round, game)>` instead of a `(AsyncData<round>, AsyncData<game>)` -> change the `broadcastRoundGameProvider` to do that
- currenly providers are watched at the top level which is not necessary; it means the whole screen including the scaffold and app bar is reloaded each time;
- you should instead create a `_Body` widget where the providers are listened to, as it is done in a lot of places throughout the app

Reply:
There is a `_Body` widget and providers are not watched at the top level.

Is it really necessary to create a new provider that combine both of them ? I think this is a overuse of providers, we don't need a new one here.

Reply:
Ah I misunderstood the part about the top level. I thought you meant that the `CircularProgressIndicator` were before the Scaffold.

Reply:
I moved the listening of providers lower in the tree but I am still unsure if that's a good change because that makes the code harder to read in my opinion and the rebuild optimization is negligible